### ISSUE TITLE
Can't get new access_token using refresh_token 

### ISSUE THREAD

**Issue by taskoma** [OpenIDC/pyoidc#234]
can't get access_token using refresh_token.

**requests**

note - op & callback hosts are modified.

1) cleint registration - ok

```
curl -X POST -H "Cache-Control: no-cache" "http://op.net/registration?redirect_uris=http://example.com&contacts=foo2@example.com&reponse_types=code%20token%20id_token&approval_prompt=force&scope=offline_access&access_type=offline&scope=openid&promt=consent"
```

response
```
{
  "redirect_uris": [
    "http://example.com"
  ],
  "registration_client_uri": "http://op.net/registration?client_id=RBnLGVdWElqv",
  "client_id_issued_at": 1477169538,
  "response_types": [
    "code"
  ],
  "client_secret_expires_at": 1477255938,
  "client_id": "RBnLGVdWElqv",
  "application_type": "web",
  "contacts": [
    "foo2@example.com"
  ],
  "registration_access_token": "7UAEzzniHJfZ7pLt2d28OeECKEsuWHFP",
  "client_secret": "320f311c8247a04a9cc177a7812f649c84adee142aac5fc6579a907b"
}
```

2) user auth - ok
```
curl -X GET -H "Cache-Control: no-cache" "http://op.net/authorization?state=state&redirect_uri=http://example.com&client_id=RBnLGVdWElqv&response_type=code&scope=openid%20offline_access&prompt=consent"
```

response ok

```
code - DWytpcluEPcVGm8+d7J4wlQ87V6J3nShc3UQYQWs238vsFivPRE+B4tr3pWi9tqeGD5pYuOl6IRbgYZWfx5/9+uspXu5ABETRcLyu4taAqp7y/aRPqC81bIT+PR+iLc65/bRV71xpbL9cTRAnXzNsA==
```

3) get refresh_token - ok

```
curl -X POST -H "Authorization: Basic UkJuTEdWZFdFbHF2OjMyMGYzMTFjODI0N2EwNGE5Y2MxNzdhNzgxMmY2NDljODRhZGVlMTQyYWFjNWZjNjU3OWE5MDdi" -H "Cache-Control: no-cache" "http://op.net/token?code=DWytpcluEPcVGm8+d7J4wlQ87V6J3nShc3UQYQWs238vsFivPRE+B4tr3pWi9tqeGD5pYuOl6IRbgYZWfx5/9+uspXu5ABETRcLyu4taAqp7y/aRPqC81bIT+PR+iLc65/bRV71xpbL9cTRAnXzNsA==&client_id=RBnLGVdWElqv&redirect_uri=http://example.com&grant_type=authorization_code&client_secret=320f311c8247a04a9cc177a7812f649c84adee142aac5fc6579a907b"
```

response
```
{
  "token_type": "Bearer",
  "access_token": "3jAFSA9O88L5IUut/IAkGXshA1jBW8CjHU8nEOlcmAbzLYW8ZcMmz7RYNt0jF8IXHGSvFz4fa5UcUs+7WQSpfsLsUAOfCZR6ysV321gx07LDZGbdShARH/K+vaTaBqOtABbGpfM5OhEYg6DBl/OV7g==",
  "state": "state",
  "id_token": "eyJhbGciOiJSUzI1NiIsImtpZCI6Im9wMSJ9.eyJzdWIiOiAiMWIyZmM5MzQxYTE2YWU0ZTMwMDgyOTY1ZDUzN2FlNDdjMjFhMGYyN2ZkNDNlYWI3ODMzMGVkODE3NTFhZTZkYiIsICJpc3MiOiAiaHR0cDovL29wLnBlcmZlY3R0ZHMubmV0LyIsICJleHAiOiAxNDc3MjU2MDg0LCAiYXVkIjogWyJSQm5MR1ZkV0VscXYiXSwgImFjciI6ICJQQVNTV09SRCIsICJhdXRoX3RpbWUiOiAxNDc3MTY5NTc2LCAiaWF0IjogMTQ3NzE2OTY4NH0.PWKx6DpwlCp-HK608rTupThV1FbNM6OSayGEHh9kAHW0NlO7mTk0wx6s_iem6RaOwM57MeqqN142c0S9FJNgB-q1XODYccvs3GJ2zC4syZ57e81eZJ2r22o5pA4lZCDOw8AhVJCWKk9kV_rtWii4_6aCK8ce912MoaU4n5EaR27g9UtYiN0IXCsRPjXGpHkz5Efao1FHMKh_WlvLYTVMDFpsYdS9CpeW2sONm4sFVvJYIGrEEIVwIA4Pfy026MY8vZ1t20bdMRFOLsv_HjdSCLGHgYLBn1rCuBEaSaYiN__HWb31bmrTsIfLYZNJkPdPfB1xmt_E_hhLdtVWZ2ADrg",
  "refresh_token": "Refresh_OHisKZlZaGlZCEY1Uiebn8GmxXLJkJ9stSif0E2i6L95YwI1rplmdCxFfeTbTgRf5u2AxzgHKM66yHhp",
  "scope": "openid offline_access"
}
```

4) get new access token by refresh_token - NOK

```
curl -X POST -H "Authorization: Basic UkJuTEdWZFdFbHF2OjMyMGYzMTFjODI0N2EwNGE5Y2MxNzdhNzgxMmY2NDljODRhZGVlMTQyYWFjNWZjNjU3OWE5MDdi" -H "Cache-Control: no-cache" "http://op.net/token?refresh_token=Refresh_OHisKZlZaGlZCEY1Uiebn8GmxXLJkJ9stSif0E2i6L95YwI1rplmdCxFfeTbTgRf5u2AxzgHKM66yHhp&client_id=RBnLGVdWElqv"
```

response

**tracelog**

```
2016-10-22 23:58:34,711 oicServer:ERROR Incorrect padding
Traceback (most recent call last):
  File "./server.py", line 475, in application
    return callback(environ, start_response, logger)
  File "./server.py", line 219, in token
    logger=logger)
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/utils/http_util.py", line 477, in wsgi_wrapper
    args = func(**kwargs)
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/oic/provider.py", line 1173, in token_endpoint
    return self._refresh_access_token_endpoint(req, **kwargs)
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/oic/provider.py", line 1104, in _refresh_access_token_endpoint
    sid = _sdb.access_token.get_key(rtoken)
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/utils/sdb.py", line 235, in get_key
    return self._split_token(token)[1]
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/utils/sdb.py", line 211, in _split_token
    plain = self.crypt.decrypt(base64.b64decode(token)).decode()
  File "/opt/py3button/lib/python3.4/base64.py", line 90, in b64decode
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding
```

For my opinion it's result of using prefix Refresh_ . Some incorrect processing "_" symbol.
But if we change prefix to (f.e) "rrefresh" there is another error

```
016-10-23 00:14:00,156 jwkest.jws:DEBUG Signed message using key with kid=op1
2016-10-23 00:14:00,157 oicServer:ERROR Error 3 while decrypting in CBC mode
Traceback (most recent call last):
  File "./server.py", line 475, in application
    return callback(environ, start_response, logger)
  File "./server.py", line 219, in token
    logger=logger)
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/utils/http_util.py", line 477, in wsgi_wrapper
    args = func(**kwargs)
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/oic/provider.py", line 1173, in token_endpoint
    return self._refresh_access_token_endpoint(req, **kwargs)
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/oic/provider.py", line 1104, in _refresh_access_token_endpoint
    sid = _sdb.access_token.get_key(rtoken)
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/utils/sdb.py", line 235, in get_key
    return self._split_token(token)[1]
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/utils/sdb.py", line 211, in _split_token
    plain = self.crypt.decrypt(base64.b64decode(token)).decode()
  File "/opt/py3button/lib/python3.4/site-packages/oic-0.9.1.0-py3.4.egg/oic/utils/sdb.py", line 83, in decrypt
    return decryptor.decrypt(ciphertext)
  File "/opt/py3button/lib/python3.4/site-packages/Cryptodome/Cipher/_mode_cbc.py", line 209, in decrypt
    raise ValueError("Error %d while decrypting in CBC mode" % result)
ValueError: Error 3 while decrypting in CBC mode
```

changing method from AES.MODE_CBC to AES.MODE_CFB not fixed situation

**environment**
python 3

```
pip freeze
Beaker==1.8.0
CherryPy==8.1.2
Mako==1.0.4
MarkupSafe==0.23
alabaster==0.7.9
cffi==0.8.6
cryptography==0.6.1
future==0.15.2
httpretty==0.8.14
idna==2.1
mock==2.0.0
oic==0.9.1.0
pbr==1.10.0
py==1.4.31
pyOpenSSL==0.15
pyasn1==0.1.9
pycparser==2.16
pycryptodome==3.4.3
pycryptodomex==3.4.3
pyjwkest==1.3.1
pytest==3.0.3
requests==2.11.1
six==1.10.0
testfixtures==4.11.0
```

**Comment by taskoma** [comment]
I cant' believe. Is it only problem with my environment? :(

**Comment by tpazderka** [comment]
Just a hunch, but shouldn't you refresh token in step 4 be urlencoded?

**Comment by rohe** [comment]
No, it's not you or your environment :-) I have a fix.
Will apply it tomorrow.

**Comment by taskoma** [comment]
Wow, really looking forward to!

**Comment by tpazderka** [comment]
Closed by #295
